nlp/2021_06_18_data_augmentation/totw_nlp_dat_aug.ipynb

```diff
@@ -0,0 +1,1789 @@
+        "\n",
+        "The training size will impact the performace of a model heavily, this notebook looks into the possibilities of performing data augmentation on a NLP dataset. Data augmentation techniques are used to generate additional samples 🪄. \n",
+        "\n",
+        "Data augmentation is already standard practice in computer vision projects 👌, but can also be leveraged in multilingual NLP problems. We'll use a limited trainingset to simulate a real-world use case, where we often are constrained by the size of the available data 🤦. \n",
+        "\n",
+        "We'll focuss on using back-translation and contextual word-embedding insertions as data augmentation techniques 🤗."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "acwqGIAyU84m"
+      },
+      "source": [
+        "## 🛠️ Getting started\n",
+        "\n",
+        "The cells below will setup everything that is required to get started with data augmentation and finetuning an NLP model with the HuggingFace API."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "buIDjl_wU84m"
+      },
+      "source": [
+        "### Setup"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "v-ItvjqP4Cxn"
+      },
+      "source": [
+        "!pip install -q transformers sentencepiece datasets tokenizers nltk nlpaug "
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "Ir6zD__FU84n"
+      },
+      "source": [
+        "### Imports"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "hM-09Chsj7wI"
+      },
+      "source": [
+        "import re\n",
+        "import numpy as np\n",
+        "import pandas as pd \n",
+        "\n",
+        "import nltk\n",
+        "import nlpaug.flow as naf\n",
+        "import nlpaug.augmenter.word as naw\n",
+        "import plotly.graph_objects as go\n",
+        "from transformers import pipeline, AutoTokenizer, AutoModelForSequenceClassification, TrainingArguments, Trainer, TrainerCallback\n",
+        "from datasets import load_dataset, concatenate_datasets, load_from_disk, load_metric"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "gFQ6LEgjeRWb"
+      },
+      "source": [
+        "### Download dataset\n",
+        "Since we're particulary interested in multilingual NLP, we'll use a well-known dutch dataset [DBRD](https://github.com/benjaminvdb/DBRD). The dataset contains over 110k book reviews along with associated binary sentiment polarity labels. The downstream task will be assigning a sentiment to a book review.  "
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "1ltsurmvaaGd"
+      },
+      "source": [
+        "max_input_len=128\n",
```

Comment:
I just realized we don't explain why we limit the input_len to 128 here, can you provide an extra comment on this?